app/gui/view/graph-editor/src/component/node/growth_animation.rs

```diff
@@ -0,0 +1,99 @@
+//! Edited node growth/shrink animation implementation.
+//!
+//! When the user starts editing of the node - it smoothly growth in size to match the 1.0x zoom
+//! factor size. After editing is finished, the node smothly shrinks to its original size. This is
+//! implemented by using a separate `edited_node` camera that is moved in synchronization with
+//! `node_searcher` camera.
+
+use ensogl::prelude::*;
+
+use crate::GraphEditorModelWithNetwork;
+use crate::NodeId;
+use enso_frp as frp;
+use ensogl::animation;
+use ensogl::display::Scene;
+use ensogl::Animation;
+use ensogl::Easing;
+
+/// Initialize edited node growth/shrink animator. It would handle scene layer change for the edited
+/// node as well.
+pub fn initialize_edited_node_animator(
+    model: &GraphEditorModelWithNetwork,
+    frp: &crate::Frp,
+    scene: &Scene,
+) {
+    let network = &frp.network;
+    let out = &frp.output;
+    let searcher_cam = scene.layers.node_searcher.camera();
+    let edited_node_cam = scene.layers.edited_node.camera();
+    let main_cam = scene.layers.main.camera();
+
+    let growth_animation = Animation::new(network);
+    let animation_blending = Easing::new(network);
+
+    frp::extend! { network
+        let searcher_cam_frp = searcher_cam.frp();
+        let main_cam_frp = main_cam.frp();
+
+
+        // === Starting  node editing. ===
+
+        previous_edited_node <- out.node_editing_started.previous();
+        _eval <- all_with(&out.node_editing_started, &previous_edited_node, f!([model] (current, previous) {
+            move_node_to_main_layer(&model, *previous);
+            move_node_to_edited_node_layer(&model, *current);
+        }));
+
+
+        // === Edited node camera position animation. ===
+
+        is_growing <- bool(&out.node_editing_finished, &out.node_editing_started);
+        edited_node_cam_target <- switch(&is_growing, &main_cam_frp.position, &searcher_cam_frp.position);
+        growth_animation.target <+ edited_node_cam_target;
+
+        on_node_editing_start_or_finish <- any(&out.node_editing_started, &out.node_editing_finished);
+        eval_ on_node_editing_start_or_finish ({
+            animation_blending.stop_and_rewind(0.0);
+            animation_blending.target(1.0);
+        });
+
+        edited_node_cam_position <- all_with3(&edited_node_cam_target,
```

Comment:
This is cool, however, this lacks detailed documentation of how `animation_blending` works. This documentation should be talkative and should explain how it is used to keep the camera and node synchronised without lags and why it is used in the first place. The Graph editor lacks the same docs and I am sad that we did not add them back then. However, in order not to produce more undoceumented code, please add it here.

Reply:
Added a documentation